core/vm/evm.go

```diff
@@ -450,6 +452,12 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if evm.StateDB.GetNonce(address) != 0 || (contractHash != (common.Hash{}) && contractHash != emptyCodeHash) {
 		return nil, common.Address{}, 0, ErrContractAddressCollision
 	}
+	// Check that [caller] has the right to deploy contracts
+	modifyStatus := getAllowListStatus(evm.StateDB, caller.Address())
```

Comment:
can we extend this to the origin as well? Otherwise, you'll need to give factories deployer privilege which seems sub-optimal.

Reply:
I'm looking around rn

Reply:
Figured out a way 🚀 ...one sec

Reply:
https://github.com/ava-labs/subnet-evm/pull/23